File: src/Java/StructuralPatterns/Facade/Multimedia/src/Facade/MultimediaFacade.java
package Facade;

import SubSystems.*;

public class MultimediaFacade {
    private final VideoPlayer videoPlayer = new VideoPlayer();
    private final AudioPlayer audioPlayer = new AudioPlayer();
    private final Display display = new Display();
    private final SubtitlesService subtitlesService = new SubtitlesService();
    private final StreamingService streamingService = new StreamingService();

    public void watchMovie(String movie, String audioTrack, String subtitlesLanguage) {
        streamingService.streamContent(movie);
        videoPlayer.playVideo(movie);
        audioPlayer.playAudio(audioTrack);
        subtitlesService.displaySubtitle(subtitlesLanguage);
        display.show("Enjoy the movie on the big screen!");
    }

    public void listenToMusic(String song) {
        streamingService.streamContent(song);
        audioPlayer.playAudio(song);
        display.show("Feel the music with visualizations on the display.");
    }
}

File: src/C#/StructuralPatterns/Facade/Multimedia/Facade/MultimediaFacade.cs
using Multimedia.SubSystems;
using System;
using System.Collections.Generic;
using System.Text;

namespace Multimedia.Facade
{
    public class MultimediaFacade
    {
        private readonly VideoPlayer videoPlayer = new VideoPlayer();

        private readonly AudioPlayer audioPlayer = new AudioPlayer();

        private readonly Display display = new Display();

        private readonly SubtitlesService subtitleService = new SubtitlesService();

        private readonly StreamingService streamingService = new StreamingService();

        public void WatchMovie(string movie, string audioTrack, string subtitlesLanguage)
        {
            streamingService.StreamContent(movie);
            videoPlayer.PlayVideo(movie);
            audioPlayer.PlayAudio(audioTrack);
            subtitleService.DisplaySubtitle(subtitlesLanguage);
            display.Show("Enjoy the movie on the big screen!");
        }

        public void ListenToMusic(string song)
        {
            streamingService.StreamContent(song);
            audioPlayer.PlayAudio(song);
            display.Show("Feel the music with visualizations on the display.");
        }
    }
}
